src/util/misc.ts

```diff
@@ -65,6 +65,35 @@ function applyNTimes<T>(func: (val: T) => T, times: number, start: T): T {
     return result;
 }
 
+/**
+ * Applies a delta change to the value of an input element
+ *
+ * @param input
+ * @param delta
+ * @returns the new value of the input field
+ */
+function applyDeltaToInput(
+    input: HTMLInputElement,
+    delta: number,
+    min: number = 0,
+    max: number = 0,
+    triggerChange: boolean = true,
+): string {
+    const oldValue = Number(input.value) || min;
+
+    if (max > 0) {
+        input.value = String(Math.clamp(oldValue + delta, min, max));
+    } else {
+        input.value = String(Math.min(oldValue + delta, min));
+    }
+
+    if (triggerChange) {
+        $(input).trigger("change");
```

Comment:
This trigger does not appear to be anywhere else in the system, but the base Item sheet does programmatically dispatch a change event ([here](https://github.com/foundryvtt/pf2e/blob/cfbf5d3a2e9f4035e94375750126b9e9cdc8f032/src/module/item/base/sheet/sheet.ts#L344-L345)). I want to clarify that it needs to be done this way - or a standard Event *with bubbling enabled* - because of the way jQuery is internally (in the Foundry form application) setup to catch the data submission and actually apply the value changes to the data model.

Reply:
We have a sheet helpers you could probably stick this in, but if this is possible with a regular event ideally use that instead, we're trying to get off jquery (core will eventually do so as well).

I'm wondering how well throttling the change would work, so that repeated hits down submit immediately and it behaves "smoother". Maybe a future consideration.

Reply:
> We have a sheet helpers you could probably stick this in, but if this is possible with a regular event ideally use that instead, we're trying to get off jquery (core will eventually do so as well).

I think I got it in the right place now in `src/module/actor/sheet/helpers.ts`

> I'm wondering how well throttling the change would work, so that repeated hits down submit immediately and it behaves "smoother". Maybe a future consideration.

I added a 50ms debounce to the event, but that was too slow for "small" fields like HP. A 0ms debounce, like I have shown now, is enough to push it to the bottom of the call stack and provides a smoother UX but is still fast enough to process changes correctly.